### ISSUE TITLE
Clicking on Iteration freezes the UI

### ISSUE THREAD

**Issue by Raunak1203** [openshiftio/openshift.io#4309]
Click on any Iteration in the side panel freezes the UI

**Comment by kwk** [comment]
@Raunak1203 what is the state of this bug? Is there an cause analysis or a PR to fix this?

**Comment by jarifibrahim** [comment]
Observation (on prod-preview and prod)

All newly created workitems are assigned to the **FIRST iteration** in the list of iterations instead of the root iteration.The payload being sent from the frontend **contains ID of the first iteration instead of the root iteration**.

Not sure if this could cause any problems.

**Comment by sahil143** [comment]
> All newly created work items are assigned to the FIRST iteration in the list of iterations instead of the root iteration. The payload being sent from the frontend contains ID of the first iteration instead of the root iteration.

@jarifibrahim can you tell me the steps for this. I'm not able to reproduce this problem. For me all newly created WorkItems have root iteration assigned to them on prod-preview.

**Comment by jarifibrahim** [comment]
>@jarifibrahim can you tell me the steps for this. I'm not able to reproduce this problem. For me all newly created WorkItems have root iteration assigned to them on prod-preview.

@sahil143 This can't be reproduce on prod-preview. My bad. But it can be reproduced on prod.
Try this
1. Make sure you have atleast one iteration
2. Create a workitem. Notice the iteration in quick preview. It should be the **root iteration**.
3. Click on the iteration in side panel
4. Go back to the main list. (click on `workitems` in case of Agile space). 
5. Create a workitem. Notice the iteration in quick preview. It should be the **first iteration in the list**.

**Comment by jarifibrahim** [comment]
>@kwk, @sahil143 is working on it but is blocked with build issues

@Raunak1203 we have a PR https://github.com/fabric8-ui/fabric8-planner/pull/2788 ready to fix build issues but we cannot merge it untill this issue is fixed. The PR build (e2e tests) would fail without the fix for this issue..

**Comment by jarifibrahim** [comment]
Looks like the issue was introduced by https://github.com/fabric8-services/fabric8-wit/commit/7ab24ac099a9316843c3044d1ad80dc4e0758934 

Steps to reproduce
1. Pull both the docker-compose files from https://gist.github.com/jarifibrahim/4a0f3e6fc6f510b38a7b3090496a2f79
2. Run `docker-compose -f docker-compose-bad.yml up` (contains https://github.com/fabric8-services/fabric8-wit/commit/7ab24ac099a9316843c3044d1ad80dc4e0758934). This would start `fabric8-ui` on `localhost:5000`.  Create an iteration and notice **UI freezes** on clicking on the iteration in side panel.
3. Run `docker rm -v -f $(docker ps -aq)` to clear up the database and kill all containers.
2. Run `docker-compose -f docker-compose-good.yml up` (contains https://github.com/fabric8-services/fabric8-wit/commit/164762f67a3a7634fa4ee1e8bb55c458281803c7). This would start `fabric8-ui` on `localhost:5000`.  Create an iteration and notice **UI doesn't freeze** on clicking on the iteration in side panel.

In short, UI doesn't freeze on commits upto https://github.com/fabric8-services/fabric8-wit/commit/164762f67a3a7634fa4ee1e8bb55c458281803c7 but starts freezing with commit https://github.com/fabric8-services/fabric8-wit/commit/7ab24ac099a9316843c3044d1ad80dc4e0758934 onwards.

Also, @sahil143 confirmed that the HTTP GET request to list the workitems when clicking on an iteration is not being sent from the UI.

**Comment by kwk** [comment]
> Create an iteration and notice UI freezes on clicking on the iteration in side panel.

@jarifibrahim can you post the iteration you've created here please? Then @tinakurian and I can have a look at it. I wonder if payload from `https://<YOUR-HOST>/api/iterations/<ITERATION-ID>` looks interesting, e.g. bad iteration path or resolved path.

**Comment by jarifibrahim** [comment]
>@jarifibrahim can you post the iteration you've created here please? Then @tinakurian and I can have a look at it. I wonder if payload from https://<YOUR-HOST>/api/iterations/<ITERATION-ID> looks interesting, e.g. bad iteration path or resolved path.

@kwk This was the request made from browser
```
curl 'http://localhost:8080/api/spaces/6a5d1b58-297d-43ce-a51c-c64b7a30c0cf/iterations' -H 'Origin: http://localhost:3000' -H 'Authorization: Bearer ...' -H 'Content-Type: application/json' -H 'Accept: application/json, text/plain, */*' --data-binary '
{
  "data": {
    "attributes": {
      "name": "first",
      "parent_path": "",
      "resolved_parent_path": "",
      "user_active": false,
      "active_status": false,
      "description": "",
      "state": "new"
    },
    "links": {
      "self": ""
    },
    "relationships": {
      "workitems": {
        "meta": {
          "total": 0,
          "closed": 0
        }
      }
    },
    "type": "iterations"
  }
}' --compressed
```

and this was the response received
```
{
  "data": {
    "attributes": {
      "active_status": false,
      "created-at": "2018-09-13T08:22:55.964832Z",
      "description": "",
      "name": "first",
      "parent_path": "/fd08ea36-abe7-4180-98eb-627c18c63311",
      "resolved_parent_path": "/asd",
      "state": "new",
      "updated-at": "2018-09-13T08:22:55.964832Z",
      "user_active": false
    },
    "id": "0740acba-af65-4dbe-a5ab-be66946bb17b",
    "links": {
      "related": "http://localhost:8080/api/iterations/0740acba-af65-4dbe-a5ab-be66946bb17b",
      "self": "http://localhost:8080/api/iterations/0740acba-af65-4dbe-a5ab-be66946bb17b"
    },
    "relationships": {
      "parent": {
        "data": {
          "id": "fd08ea36-abe7-4180-98eb-627c18c63311",
          "type": "iterations"
        },
        "links": {
          "related": "http://localhost:8080/api/iterations/fd08ea36-abe7-4180-98eb-627c18c63311",
          "self": "http://localhost:8080/api/iterations/fd08ea36-abe7-4180-98eb-627c18c63311"
        }
      },
      "space": {
        "data": {
          "id": "6a5d1b58-297d-43ce-a51c-c64b7a30c0cf",
          "type": "spaces"
        },
        "links": {
          "related": "http://localhost:8080/api/spaces/6a5d1b58-297d-43ce-a51c-c64b7a30c0cf",
          "self": "http://localhost:8080/api/spaces/6a5d1b58-297d-43ce-a51c-c64b7a30c0cf"
        }
      },
      "workitems": {
        "links": {
          "related": "http://localhost:8080/api/workitems/?filter[iteration]=0740acba-af65-4dbe-a5ab-be66946bb17b"
        },
        "meta": {
          "closed": 0,
          "total": 0
        }
      }
    },
    "type": "iterations"
  }
}
```
